File: Presentation/ClientApp/src/app/models/project.ts
import { Task } from "./task";

export interface Project {
    id: string;
    title: string;
    description: string;
    tasks: Task[] | null;
    accessRights: AccessRights
}

export enum AccessRights {
  Read = 1,
  Write = 2,
  Owner = 3
}

File: Domain/Entities/Project.cs
using Domain.Common;

namespace Domain.Entities
{
    public class Project : BaseAuditableEntity
    {
        public required string Title { get; set; }

        public required string Description { get; set; }

        public ICollection<Task> Tasks { get; private set; } = new List<Task>();
    }
}
